huaweicloud/services/acceptance/cc/data_source_huaweicloud_cc_network_instances_test.go

```diff
@@ -0,0 +1,166 @@
+package cc
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDataSourceCcNetworkInstances_basic(t *testing.T) {
+	dataSource := "data.huaweicloud_cc_network_instances.filter_by_id"
+	rName := acceptance.RandomAccResourceName()
+	dc := acceptance.InitDataSourceCheck(dataSource)
+	baseConfig := testDataSourceCcNetworkInstances_base(rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourceCcNetworkInstances_basic(baseConfig, rName),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(dataSource, "network_instances.0.id"),
+					resource.TestCheckResourceAttrSet(dataSource, "network_instances.0.instance_id"),
+					resource.TestCheckResourceAttrSet(dataSource, "network_instances.0.cloud_connection_id"),
+					resource.TestCheckOutput("is_id_useful", "true"),
+					resource.TestCheckOutput("is_cloud_connection_id_filter_useful", "true"),
+					resource.TestCheckOutput("is_description_filter_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testDataSourceCcNetworkInstances_base(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_vpc" "test1" {
+  name = "%[1]s1"
+  cidr = cidrsubnet("10.12.0.0/16", 4, 1)
+}
+
+resource "huaweicloud_vpc" "test2" {
+  name = "%[1]s2"
+  cidr = cidrsubnet("192.168.0.0/16", 4, 1)
+}
```

Comment:
```suggestion
resource "huaweicloud_vpc" "test1" {
  count = 2
  name = "%[1]s_${count.index}"
  cidr = cidrsubnet("10.12.0.0/16", 4, 1)
}
```
fix huaweicloud_vpc_subnet, huaweicloud_cc_network_instance like this

Reply:
fixed